forge-game/src/main/java/forge/game/ability/effects/AddPhaseEffect.java

```diff
@@ -20,8 +23,14 @@ public class AddPhaseEffect extends SpellAbilityEffect {
 
     @Override
     public void resolve(SpellAbility sa) {
-        boolean isTopsy = sa.getActivatingPlayer().getAmountOfKeyword("The phases of your turn are reversed.") % 2 == 1;
-        PhaseHandler phaseHandler = sa.getActivatingPlayer().getGame().getPhaseHandler();
+        final Card host = sa.getHostCard();
+        final Player activator = sa.getActivatingPlayer();
+        boolean isTopsy = activator.getAmountOfKeyword("The phases of your turn are reversed.") % 2 == 1;
+        PhaseHandler phaseHandler = activator.getGame().getPhaseHandler();
+
+        //You can only add extra phases to your turn
+        if (!phaseHandler.isPlayerTurn(activator)) return;
```

Comment:
that check is way too broad
you need to implement it by the rules:
> 500.10a If an effect that says “you get” an additional step or phase would add a step or phase to a turn other than its controller’s, no steps or phases are added.

Reply:
nooo
this card is the only one that says that
so please don't "break" all others :P

Reply:
technically all others also would just give you extra phases on your turns

so checking for activating player should be okay

Reply:
@Hanmac 
no, check e. g. _Relentless Assault_
>  If you manage to cast this during a main phase of your opponent's turn, that opponent's creatures will untap and that opponent will be able to attack again. It will not allow you to attack during their turn.

Reply:
It kinda sounds like we need a param to specify ownership of the phase, and return/continue if it doesn't fit?

Reply:
or `ConditionPlayerTurn$`